express/component_spec.py

```diff
@@ -0,0 +1,346 @@
+"""This module defines classes to represent an Express component specification."""
+import copy
+import json
+import yaml
+import pkgutil
+import types
+import typing as t
+from pathlib import Path
+from dataclasses import dataclass, asdict
+
+import jsonschema.exceptions
+from jsonschema import Draft4Validator
+from jsonschema.validators import RefResolver
+
+from express.exceptions import InvalidComponentSpec
+from express.io_utils import load_yaml
+from express.common import Field
+from express.pipeline_utils import get_kubeflow_type
+
+
+@dataclass
+class KubeflowInput:
+    """
+    Kubeflow component input argument
+    Attributes:
```

Comment:
This should be `Args` instead of `Attributes`. We follow the google docstring format.

Reply:
Same for a lot of docstrings in this PR.